FluentTerminal.App/Views/NativeFrontend/TerminalControl.xaml.cs

```diff
@@ -424,14 +457,19 @@ private void ProcessTerminalCode(TerminalCode code)
 
         private void Terminal_OutputReceived(object sender, byte[] e)
         {
+  
             var text = Encoding.UTF8.GetString(e, 0, e.Length);
             var reader = new ArrayReader<char>(text.ToCharArray());
             var codes = _parser.Parse(reader);
             foreach (var code in codes)
             {
                 ProcessTerminalCode(code);
             }
-            canvas.Invalidate();
+
```

Comment:
we might need to wrap this in a dispatcher. You can see some dispatcher usage in XtermView